source/extensions/io_socket/io_uring/io_handle_impl.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "envoy/network/io_handle.h"
+
+#include "source/common/common/logger.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace IoSocket {
+namespace IoUring {
+
+class IoUringFactory;
+struct Request;
+
+/**
+ * IoHandle derivative for sockets.
+ */
+class IoUringSocketHandleImpl final : public Network::IoHandle,
```

Comment:
nit: you can inherit from the concrete impl of `IoSocketHandleImpl` if you want to reduce the code

Reply:
I wanted to avoid inheritance as the type of the relationship doesn't feel like "is-a" to me.